node/flatpak_node_generator/providers/npm.py

```diff
@@ -110,7 +110,7 @@ def _process_packages_v2(
             source: PackageSource
             package_json_path = lockfile.parent / install_path / 'package.json'
             if (
-                'node_modules' not in package_json_path.parents
+                Path('node_modules') not in package_json_path.parents
```

Comment:
OOF that's a nasty bug! If the intent is to see if *any* component is `node_modules` though, this also doesn't work:

```python3
>>> Path('node_modules') in Path('node_modules/x').parents
True
>>> Path('node_modules') in Path('a/node_modules/x').parents
False
>>>
```

Perhaps this should instead be:

```suggestion
                'node_modules' not in package_json_path.parts[:-1]
```

(not actually sure if the `:-1` is needed to trim off the last part; the OG code used `parents`, but I dunno how important excluding the final component is)

Reply:
Good catch - yes I'm not exactly sure if we're just trying to catch any dir called `node_modules` in the path or if it has to be the last/first. Maybe I'll see if there's a spec for this or something in the node docs.

Reply:
Yeah this definitely should've been `package_json_path.parts`. My bad.

Reply:
That works for me, I've made the change.